src/fpm_filesystem.F90

```diff
@@ -323,12 +335,25 @@ function read_lines(fh) result(lines)
     integer, intent(in) :: fh
     type(string_t), allocatable :: lines(:)
 
-    integer :: i
-    integer :: iostat
+    integer :: i, length
+    character(len=:), allocatable :: content
+    integer, allocatable :: first(:), last(:)
+
+    inquire (fh, size=length)
+    allocate (character(len=length) :: content)
+    if (length == 0) then
+        allocate (lines(0))
+        return
+    end if
+
+    ! read file into a single string
+    read (fh) content
+    call split_first_last(content, c_new_line, first, last)  ! TODO: \r (< macOS X), \n (>=macOS X/Linux/Unix), \r\n (Windows)
```

Comment:
This PR looks great, the only comment I have is whether all possible new line configurations are properly handled in the bulk binary read. All CI tests are passed, but are we sure we aren't missing anything?

Reply:
@perazz Thanks for reviewing.

I'm not familiar with macOS, and the main purpose of this PR is to improve IO speed by reading files as binary instead of line by line; but I eventually realized that once we read text files as binary, then we need to deal with `EOL` (end of line) operations, which is related to the current design of fpm.

I just subconsciously went and looked at file reading in meson.build, and realized that [it also reads bytes or characters](https://github.com/mesonbuild/meson/blob/059b9e048d3b69f3aa62c559e9c0da8c7b943bfd/mesonbuild/interpreterbase/interpreterbase.py#L138), not lines, and that [it considers line breaks `EOL` to be `\n` and doesn't talk about `\r` separately](https://github.com/mesonbuild/meson/blob/059b9e048d3b69f3aa62c559e9c0da8c7b943bfd/mesonbuild/mparser.py#L115C15-L115C15), in which case the difficulty goes right down.